synfig-studio/src/gui/dialogs/dialog_pluginmanager.cpp

```diff
@@ -0,0 +1,298 @@
+        plugin_list_item->set_margin_top(10);
+        plugin_option_box->property_expand().set_value(false);
+
+        restore_settings->set_image_from_icon_name("view-refresh", Gtk::ICON_SIZE_BUTTON);
+        open_folder->set_image_from_icon_name("document-open", Gtk::ICON_SIZE_BUTTON);
+        delete_plugin->set_image_from_icon_name("user-trash-symbolic", Gtk::ICON_SIZE_BUTTON);
+
+        delete_plugin->signal_clicked().connect([plugin, this](){
+            this->message_dialog.set_message(_("Do you want to delete the plugin: ") + plugin.name.get());
+            int response = this->message_dialog.run();
+            if (response == Gtk::RESPONSE_OK) {              
+                App::plugin_manager.remove_plugin(plugin.id);
+            }
+            this->message_dialog.close();
+        });
+
+        open_folder->signal_clicked().connect([plugin](){
+            plugin.launch_dir();
+        });
+
+        plugin_option_box->pack_start(*restore_settings, Gtk::PACK_SHRINK, 10);
+        plugin_option_box->pack_start(*open_folder, Gtk::PACK_SHRINK, 10);
+        plugin_option_box->pack_start(*delete_plugin, Gtk::PACK_SHRINK, 10);
+        plugin_option_box->set_margin_top(10);
+        plugin_option_box->set_margin_bottom(10);
+
+        plugin_name->set_text(plugin.name.get());
+
+        plugin_list_item->pack_start(*plugin_name, Gtk::PACK_SHRINK, 20);
+        plugin_list_item->pack_end(*plugin_option_box, Gtk::PACK_SHRINK, 20);
+        
+        plugin_list_box.add(*plugin_list_item);
+
+    }
+    plugin_list_box.show_all_children();
+}
+
+std::string extract_plugin_name(std::istream &fileStream)
+{
+    xmlpp::DomParser parser;
+    parser.set_substitute_entities();
+    parser.parse_stream(fileStream);
+
+    const xmlpp::Node* pNode = parser.get_document()->get_root_node();
+    if (std::string(pNode->get_name()) != std::string("plugin") ) {
+        synfig::warning("Invalid plugin.xml file (missing root <plugin>)");
+        return std::string("");
+    }
+
+    auto execlist = pNode->find("./exec");
+    if (!execlist.empty())
+    {
+        for (const xmlpp::Node* node : pNode->find("./name") ) {
+            const xmlpp::Element* element = dynamic_cast<const xmlpp::Element*>(node);
+            std::string lang = element->get_attribute_value("lang");
+            if (lang.empty()) {
+                return element->get_child_text()->get_content();
+            }
+        }
+    }
+}
+void
+Dialog_PluginManager::on_install_plugin_button_clicked()
+{
+    // Show the dialog and wait for a user response
+    int result = plugin_file_dialog.run();
+    switch (result) {
+        case (Gtk::RESPONSE_OK):
+        { 
+            std::vector<std::string> files;
+            PluginZipStatus pluginStatus = PluginZipStatus::PLUGIN_ZIP_NO_PLUGIN_XML;
+            etl::handle<synfig::FileSystemNative> native_fs = FileSystemNative::instance();
+            etl::handle<synfig::FileContainerZip> zip_fs = new FileContainerZip();
```

Comment:
Prefer `synfig::FileSystemNative::Handle` and `synfig::FileContainerZip::Handle`.